bigquery_etl/view/publish_views.py

```diff
@@ -135,8 +153,12 @@ def main(**kwargs):
     for target in args.target:
         if os.path.isdir(target):
             for root, dirs, files in os.walk(target):
-                if "view.sql" in files:
-                    sql_files.append(os.path.join(root, "view.sql"))
+                dataset_dir = os.path.dirname(root)
```

Comment:
One thing to note is that I'm not sure which places in this repo leverage the actual directory structure as source of truth (https://github.com/mozilla/bigquery-etl/pull/1663 being a case where the directory was not a source of truth). `publish_views` I think until this PR did not rely on directory structure and instead used the view content in e.g. the logic for deciding whether to treat a view specially in conjunction with `--target-project`. It may be preferable to leverage the same filtering mechanism for both operations, or to enforce (perhaps in CI) a constraint that makes sure directory structure matches the (authoritative) view content.

This said, the main point of a flag like this is so that downstream logic doesn't have to worry about things like directory structure as much, and I think this will do the right thing.

Reply:
Having CI check would probably be the easiest way and could at the same time solve https://github.com/mozilla/bigquery-etl/issues/1413

I think in the future we'll want to specify in the `metadata.yaml` file whether a view is user-facing or not instead of relying on the dataset name. This way we publishing views won't depend on the directory structure and we don't have to maintain the list of suffixes.

Reply:
> I think in the future we'll want to specify in the metadata.yaml file whether a view is user-facing or not instead of relying on the dataset name. This way we publishing views won't depend on the directory structure and we don't have to maintain the list of suffixes.

At some point the plan is to externalize the [datasets metadata](https://github.com/mozilla-services/cloudops-infra/blob/master/projects/data-shared/tf/prod/envs/prod/bigquery/namespaces.tfvars.json) to bq-etl or MPS or similar. I haven't quite figured out what this looks like, but something to note here is that whether a view is user-facing is mostly a property of the dataset containing the view and not the view itself. A system where we have dataset metadata that propagates to individual views would likely be easiest to reason about.

See https://docs.google.com/document/d/1ug7bP7-WJ6w0WypjDoSDWEdwav7suyJ6ECjGVIcRVgE/edit# for some preliminary design around this.